deepmd/_vendors/ndtensorflow/_array.py

```
@@ -0,0 +1,531 @@
+    if rank == 0:
+        return tf.where(key, tf.broadcast_to(value, tf.shape(tensor)), tensor)
+    tensor_rank = tensor.shape.rank
+    if tensor_rank is None:
+        raise IndexError("indexed array rank must be statically known")
+    if rank > tensor_rank:
+        raise IndexError("boolean index shape is incompatible with indexed array")
+    tensor_shape = tensor.shape.as_list()
+    key_shape = key.shape.as_list()
+    if any(
+        tensor_dim is not None and key_dim is not None and tensor_dim != key_dim
+        for tensor_dim, key_dim in zip(tensor_shape, key_shape)
```

Ruff (0.15.18)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)